src/map/utils/battleutils.cpp

```diff
@@ -1381,10 +1381,10 @@ namespace battleutils
 
                 if (PAttacker->objtype == TYPE_PC && PAttacker->PParty != nullptr)
                 {
-                    if (auto* PLeader = dynamic_cast<CCharEntity*>(PAttacker->PParty->GetLeader()))
+                    if (auto* PChar = dynamic_cast<CCharEntity*>(PAttacker))
                     {
-                        PLeader->ForPartyWithTrusts([&](CBattleEntity* PMember)
-                                                    {
+                        PChar->ForPartyWithTrusts([&](CBattleEntity* PMember)
+                                                  {
```

Comment:
While you're here, can you wrap this in // clang-format on/off and align this lambda please?